src/main/kotlin/org/rust/ide/refactoring/move/common/RsMoveRetargetReferencesProcessor.kt

```diff
@@ -0,0 +1,122 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.refactoring.move.common
+
+import com.intellij.openapi.project.Project
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.RsMod
+import org.rust.lang.core.psi.ext.hasColonColon
+
+class RsMoveRetargetReferencesProcessor(
+    project: Project,
+    private val sourceMod: RsMod,
+    private val targetMod: RsMod
+) {
+
+    private val psiFactory: RsPsiFactory = RsPsiFactory(project)
+    private val codeFragmentFactory: RsCodeFragmentFactory = RsCodeFragmentFactory(project)
+
+    fun retargetReferences(referencesAll: List<RsMoveReferenceInfo>) {
+        val (referencesDirectly, referencesOther) = referencesAll
+            .partition { it.isInsideUseDirective }
+
+        for (reference in referencesDirectly) {
+            retargetReferenceDirectly(reference)
+        }
+        for (reference in referencesOther) {
+            val pathOld = reference.pathOld
+            if (pathOld.resolvesToAndAccessible(reference.target)) continue
+            val success = !pathOld.isAbsolute() && tryRetargetReferenceKeepExistingStyle(reference)
+            if (!success) {
+                retargetReferenceDirectly(reference)
+            }
+        }
+    }
+
+    private fun retargetReferenceDirectly(reference: RsMoveReferenceInfo) {
+        val pathNew = reference.pathNew ?: return
+        replacePathOld(reference, pathNew)
+    }
+
+    // "keep existing style" means that
+    // - if `pathOld` is `foo` then we keep it and add import for `foo`
+    // - if `pathOld` is `mod1::foo`, then we change it to `mod2::foo` and add import for `mod2`
+    // - etc for `outer1::mod1::foo`
+    private fun tryRetargetReferenceKeepExistingStyle(reference: RsMoveReferenceInfo): Boolean {
+        val pathOld = reference.pathOld
+        val pathNew = reference.pathNew ?: return false
+
+        val pathOldSegments = pathOld.text.split("::")
```

Comment:
@vlad20012 Could you please tell whether it's OK to use `RsPath.text` here and below (e.g. in `replacePathOld` method)? Or maybe `RsPath.referenceName` should be used instead?

Reply:
It's almost always NOT ok because of possible whitespace/comments between pats segments: `foo::/*foobar*/   bar :: baz`.

Fun fact is that we use it anyway in AutoImportFix:
```
private val RsUseItem.pathAsList: List<String>?
    get() = useSpeck?.path?.text?.split("::")
```

Reply:
Ok, thanks, so I added new method `textNormalized`, which excludes comments and whitespaces